paper-server/src/main/java/io/papermc/paper/util/PaperPoiSearchResult.java

```diff
@@ -0,0 +1,24 @@
+package io.papermc.paper.util;
+
+import io.papermc.paper.entity.PoiType;
+import org.bukkit.Location;
+
+public class PaperPoiSearchResult implements PoiSearchResult {
```

Comment:
record maybe?
Sound easier. Also nullmarked.

Reply:
This is standard to null mark implementations, so I know going forward?

Reply:
I switched this, but I still think maybe if we go record should we use record notation for methods in PoiSearchResult API e.g. location() instead of getLocation() and poiType() instead of getPoiType()? I like the record pattern more for such objects, but this really isn't the "standard" with the other result APIs